gobblin-cluster/src/main/java/org/apache/gobblin/cluster/TaskInstance.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.cluster;
+
+/**
+ * A reference to an instance of a single task that runs in a separate process.
+ */
+class TaskInstance {
+  Process taskProcess;
+
+  public TaskInstance(Process taskProcess) {
+    this.taskProcess = taskProcess;
+  }
+
+  void Cancel() {
+    this.taskProcess.destroyForcibly();
```

Comment:
why not also let TaskInstance attempt to gracefully exit?

Reply:
I am not sure if it is needed yet. 
We can discuss in person. 
We can add it later if needed.

Reply:
Anyway, this change is no longer in this PR.